slick/src/main/scala/slick/util/ReadAheadIterator.scala

```diff
@@ -49,3 +43,11 @@ trait ReadAheadIterator[+T] extends BufferedIterator[T] {
     } else throw new NoSuchElementException("next on empty iterator");
   }
 }
+
+object ReadAheadIterator {
+
+  /** Feature implemented in Scala library 2.12 this maintains functionality for 2.11 */
+  implicit class headOptionReverseCompatibility[T](readAheadIterator: ReadAheadIterator[T]){
```

Comment:
I think this can be made a [value class](http://docs.scala-lang.org/overviews/core/value-classes.html), which I believe is best practice for these kinds of extension method wrappers. Same goes for making the class `final`.